xchange-poloniex2/src/main/java/info/bitrich/xchangestream/poloniex2/dto/TradeEvent.java

```diff
@@ -9,48 +9,36 @@
  */
 public class TradeEvent {
     private String tradeId;
-    private String type;
-    private BigDecimal rate;
-    private BigDecimal amount;
-    private String date;
+    private Order.OrderType type;
+    private BigDecimal price;
+    private BigDecimal size;
+    private int timestampSeconds;
```

Comment:
You're sort of mixing source-style and target-style data here.  For clarity, it'd be better if:

* EITHER all these properties were in the source format (as they were before) and all converted when the object is mapped to an XChange object later
* OR all the properties were in the target format (in which case this field should be a `Date` rather than an `int`).

Reply:
I agree, but partially. Usage of side as "1" or "0" is very difficult, so it makes sense to convert it to something clearly understandable  asap to avoid confusion later in the code (which actually caused this bug). For example in REST api it comes as string "buy" or "sell" which caused incorrect handling.  So I treat this value still as SOURCE (in your terms).

Regarding the Date vs int. Date is different for example it has precision of milliseconds. I actually think that Date is  bad choice for timestamp, but it's in the core library so we are not going to mess with it. 
Originally the code was converting int->String then String -> Date. So here imo we also use Source value and later convert .
We can move conversion to TradeEvent and let it return both original value(string, int) and  "adapted" (OrderType, Date).
Though I think that storing "1" and "0" is a problem and will cause confusion, so we should drop them asap